test/harvest/test_wos.py

```diff
@@ -0,0 +1,214 @@
+import logging
+import os
+import re
+from dataclasses import dataclass
+
+import dotenv
+import pytest
+import requests
+
+from rialto_airflow.database import Publication
+from rialto_airflow.harvest import wos
+from rialto_airflow.snapshot import Snapshot
+from test.utils import num_jsonl_objects
+
+dotenv.load_dotenv()
+
+
+wos_key = os.environ.get("AIRFLOW_VAR_WOS_KEY")
+
+
+@pytest.fixture
+def mock_wos(monkeypatch):
+    """
+    Mock our function for fetching publications by orcid from Web of Science.
+    """
+
+    def f(*args, **kwargs):
+        yield {
+            "static_data": {
+                "summary": {
+                    "titles": {
+                        "title": [{"type": "source", "content": "An example title"}]
+                    }
+                }
+            },
+            "cluster_related": {
+                "identifiers": {
+                    "identifier": [
+                        {"type": "issn", "value": "2211-9124"},
+                        {
+                            "type": "doi",
+                            "value": "https://doi.org/10.1515/9781503624153",
+                        },
+                    ]
+                }
+            },
+        }
+
+    monkeypatch.setattr(wos, "orcid_publications", f)
+
+
+@pytest.fixture
+def mock_many_wos(monkeypatch):
+    """
+    A fixture for that returns 1000 fake documents from Web of Science.
+    """
+
+    def f(*args, **kwargs):
+        for n in range(1, 1000):
+            yield {"UID": f"mock:{n}"}
+
+    monkeypatch.setattr(wos, "orcid_publications", f)
+
+
+@pytest.fixture
+def existing_publication(test_session):
+    with test_session.begin() as session:
+        pub = Publication(
+            doi="10.1515/9781503624153",
+            sulpub_json={"sulpub": "data"},
+            wos_json={"wos": "data"},
```

Comment:
Is there a reason to have `wos_json` already populated in the existing_publication fixture? It seems like the assertion on L149 would always pass because of the fixture.

Reply:
This is a good point. I'll unset it.